pkg/networkservice/chains/nsmgrproxy/server_test.go

```diff
@@ -21,36 +21,45 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
 	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/cls"
 	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/kernel"
 	"github.com/networkservicemesh/api/pkg/api/registry"
-	"github.com/stretchr/testify/require"
-	"go.uber.org/goleak"
 
+	"github.com/networkservicemesh/sdk/pkg/registry/common/dnsresolve"
 	"github.com/networkservicemesh/sdk/pkg/tools/sandbox"
 )
 
+func testDomain(ctx context.Context, t *testing.T, dnsServer dnsresolve.Resolver) *sandbox.Domain {
+	domain := sandbox.NewBuilder(t).
+		SetNodesCount(1).
+		SetContext(ctx).
+		SetDNSResolver(dnsServer).
+		Build()
+
+	_, err := domain.Nodes[0].NewForwarder(ctx, new(registry.NetworkServiceEndpoint), sandbox.GenerateTestToken)
+	require.NoError(t, err)
+
+	return domain
+}
```

Comment:
this looks a little bit complex than before.

Reply:
Yep, it is. Fixed this with `.SetNodeSetup` and `defaultSetupNode`:
```go
// SetupNodeFunc setups each node on Builder.Build() stage
type SetupNodeFunc func(ctx context.Context, node *Node)

// SetNodeSetup replaces default node setup to custom function
func (b *Builder) SetNodeSetup(f SetupNodeFunc) *Builder {
	b.setupNode = f
	return b
}

func defaultSetupNode(t *testing.T) SetupNodeFunc {
	return func(ctx context.Context, node *Node) {
		_, err := node.NewForwarder(ctx, new(registryapi.NetworkServiceEndpoint), GenerateTestToken)
		require.NoError(t, err)
	}
}
```